app/src/main/java/org/greenstand/android/TreeTracker/fragments/LoginFragment.kt

```diff
@@ -0,0 +1,182 @@
+package org.greenstand.android.TreeTracker.fragments
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentTransaction
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.fragment_login.*
+import kotlinx.android.synthetic.main.fragment_user_identification.*
+import org.greenstand.android.TreeTracker.R
+import org.greenstand.android.TreeTracker.activities.CameraActivity
+import org.greenstand.android.TreeTracker.application.Permissions
+import org.greenstand.android.TreeTracker.utilities.ImageUtils
+import org.greenstand.android.TreeTracker.utilities.ValueHelper
+import org.greenstand.android.TreeTracker.utilities.ValueHelper.EMAIL_ADDRESS
+import org.greenstand.android.TreeTracker.utilities.ValueHelper.PHONE_NUMBER
+import org.greenstand.android.TreeTracker.viewmodels.PlanterDetailsViewModel
+import timber.log.Timber
+
+class LoginFragment : Fragment(){
+    var phoneNumberEntered: String? = null
+    var emailEntered: String? = null
+    private var mPhotoPath: String? = null
+    lateinit var viewModel: PlanterDetailsViewModel
+    private var fragment: Fragment? = null
+    private var fragmentTransaction: FragmentTransaction? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        viewModel = ViewModelProviders.of(this).get(PlanterDetailsViewModel::class.java)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_login, container, false)
+}
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+
+        requireActivity().toolbarTitle?.setText(R.string.greenstand_welcome_text)
+
+        sign_up_button?.visibility = View.INVISIBLE
+
+        phoneEditTextLogin.addTextChangedListener(object: TextWatcher {
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                inactivateLoginButton()
+            }
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                inactivateLoginButton()
+            }
+
+            override fun afterTextChanged(editTextInputs: Editable) {
+                phoneNumberEntered = editTextInputs.toString()
+                viewModel.isUserPresentOnDevice(phoneNumberEntered!!).observe(this@LoginFragment,
```

Comment:
You should not do any observing inside any of these callbacks for the edit text. This is creating a new observer every time the text changes. Instead we can make a function in the viewmodel called something like 
`fun updatePhoneNumber(phoneNumber: String)`
That can be called each time text is updated. Then, outside of any callbacks, you can observer if the user is present on the device. The method `isUserPresentOnDevice` doesn't have to take any input, since the `updatePhoneNumber` will tell the viewmodel what was input by the user.